WORKSPACE

```diff
@@ -1336,8 +1335,8 @@ go_repository(
     name = "com_github_containerd_containerd",
     build_file_proto_mode = "disable",
     importpath = "github.com/containerd/containerd",
-    sum = "h1:K2U/F4jGAMBqeUssfgJRbFuomLcS2Fxo1vR3UM/Mbh8=",
-    version = "v1.3.9",
+    sum = "h1:mdXu4r70lQky9rbOBtodOuYDDK/VKEGVK3FLdhTGkFc=",
+    version = "v1.4.6",
```

Comment:
We cannot switch to containerd 1.4 yet because we still need to support clusters using 1.3. The rest of the changes look good. Can you make it work with 1.3?

Reply:
I think it's doable, but probably not as nice to look at. :)

Either we would have to wrap `errdefs.ToGRPC` ourselves in some utility package, switch to `github.com/pkg/errors` to wrapping errors (only in the shim), or not wrap errors at all (i.e., just return the code from `errdefs`). My inclination is that the least ugly would be to use `github.com/pkg/errors` until it's safe to upgrade to 1.4, but it would introduce an additional maintenance burden to make sure no one accidentally adds `fmt.Errorf("... %w ...")` in `pkg/shim` only. What do you think?

Also, are we sure that the shim APIs aren't compatible between 1.3 and 1.4? I was under the impression that they had mostly standardized that so a 1.3 shim could work against a 1.4 server and vice versa.

Reply:
The compatibility guarantee is from container to the shim, e.g. a shim built with 1.3 will work with containerd 1.4. But the opposite is not supported, a shim built with 1.4 has no guarantee to work with containerd 1.3.

I think the best way for now is to create our own version of `errdefs.ToGRPC` that uses Go 1.13's error wrapping. Then we can replace it with `errdefs.ToGRPC` from containerd again when we upgrade to 1.4. wdyt?

Also check #6229. None of the pid files are actually needed after they have been read, so they can be removed in Start(). This should also help with the PID file leak.

Reply:
Ah, got it -- thanks for the clarification there.

I think you're right about making a custom `ToGRPC` method. Trying to remove the `%w`-style wrapping is too daunting and it will definitely work its way back in. I'll get an updated PR up in a few minutes here with that change.

#6229 looks great and it would be an even better fix to the problem that would avoid the implementation-detail handling of `Kill` before `Delete` from the containerd server.